### PR TITLE
First release!

### PR DESCRIPTION
**Pull request by avani-bhojwani**
## PR checklist

- [N/A] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [X] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/denovotranscript/tree/master/.github/CONTRIBUTING.md)
- [X] If necessary, also make a PR on the nf-core/denovotranscript _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [X] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [N/A] `CHANGELOG.md` is updated.
- [X] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: nextflow.config
Comment (fbdtemme) [inline nextflow.config:17]: Parameter names should be all `snake_case`.

```suggestion
    qc_only                    = false
```

File: workflows/denovotranscript.nf
Comment (fbdtemme) [inline workflows/denovotranscript.nf:182]: I think it would be more convenient to pass the different aligners here in a single comma separated list that you can then split into a list.

`--tools "rnaspades,trinity"`

```
def tools = params.tools.tokenize(',').

...
if (tools.contains("trinity")) {
    ....
}
```

This will scale better if you would later add more assemblers.

File: nextflow.config
Comment (fbdtemme) [inline nextflow.config:56]: All the parameters here taking an optional file should have null as the default instead of empty lists.
If you need to pass an empty list into modules/subworkflows you can use eg: `params.busco_lineages_path ?: []`

File: workflows/denovotranscript.nf
Comment (avani-bhojwani) [reply workflows/denovotranscript.nf]: thank you for the suggestion! I have added the `--assemblers` param to do this

File: CHANGELOG.md
Comment (mashehu) [inline CHANGELOG.md:6]:
```suggestion
## v1.0.0 - [2024-07-19]
```
Best to use real semantic versioning from the beginning. 

Any fun format for release names in mind?

File: docs/output.md
Comment (mashehu) [inline docs/output.md:1]:
```suggestion
# nf-core/denovotranscript: Output
```

File: CHANGELOG.md
Comment (avani-bhojwani) [reply CHANGELOG.md]: not yet! Maybe animals-flowers

**Comment by avani-bhojwani** [comment]
Thank you for reviewing! I am finished with all the changes

**Comment by avani-bhojwani** [comment]
@mashehu @fbdtemme Sorry, just waiting for your approval to go ahead with merging

File: docs/images/mqc_general_stats.png
Comment (mashehu) [inline docs/images/mqc_general_stats.png:1]: can you try to get one line per sample name (i know multiqc can be a bit tricky there, but usually it is just some slight differences in the sample names leading to the split up)

File: docs/output.md
Comment (mashehu) [inline docs/output.md:128]:
```suggestion
    the `assemblers` parameter.
```

File: docs/output.md
Comment (mashehu) [inline docs/output.md:216]:
````suggestion
```tsv title="rnaquast/*_output/basic_metrics.txt"
````

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:53]:
```suggestion
## Example workflow run
```

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:67]:
```suggestion
2. Run the pipeline with `qc_only` and any custom parameters that you have decided to use based on your data. Use `resume` to avoid unnecessarily rerunning unchanged steps
```

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:69]:
````suggestion
```bash {6-8}
````
highlights the added lines

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:83]:
```suggestion
```bash {2,7,8}
```

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:95]: something is off in this sentence after "full samplesheeet"

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:97]:
```suggestion
```bash {2,3,8}
```

File: docs/usage.md
Comment (mashehu) [inline docs/usage.md:128]:
```suggestion
- `--strand_specific` parameter can be used to set the strand-specific type for rnaSPAdes.
```
or is that parameter name coming from rnaSPAdes?

File: docs/usage.md
Comment (mashehu) [reply docs/usage.md:53]: maybe move this section to the end or if you want you can have it in a separate usage docs file (see for example https://nf-co.re/taxprofiler/1.1.8/docs/usage/tutorials)

File: modules/local/orp_transrate/main.nf
Comment (mashehu) [inline modules/local/orp_transrate/main.nf:1]: looks like this module is already in very good shape, any plans on adding it to nf-core modules?

File: modules/local/rnaquast/main.nf
Comment (mashehu) [inline modules/local/rnaquast/main.nf:1]: same here 🙂

File: nextflow.config
Comment (mashehu) [inline nextflow.config:38]:
```suggestion
    strand_specific             = null
```
if you agree with my comment before

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:79]:
```suggestion
                    "description": "Specify `true` to save files that failed to pass trimming thresholds ending in *.fail.fastq.gz"
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:9]:
```suggestion
            "title": "Partial run options",
```
or something similar more descriptive

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:87]:
```suggestion
                    "description": "Skip the fastp process if `true`"
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:91]:
```suggestion
                    "description": "Skip FastQC processes if `true`"
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:95]:
```suggestion
                    "description": "Extra arguements for fastp. For example, `--trim_front1 15 --trim_front2 15 --trim_tail1 5 --trim_tail2 5{:bash}`"
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:131]:
```suggestion
                    "description": "Assemblers to use. Possible options include `trinity`, `trinity_no_norm` (trinity without normalized reads), and `rnaspades`."
```
you could add some automatic validation of the correct assembles by providing a regex pattern in the `pattern` field for this parameter

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:139]:
```suggestion
                    "description": "Include hard filtered transcripts (in addition to medium filtered transcripts) from rnaSPAdes in the input to EvidentialGene tr2aacds."
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:143]:
```suggestion
                    "description": "Include soft filtered transcripts (in addition to the medium filtered transcripts) from rnaSPAdes in the input to EvidentialGene tr2aacds."
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:145]: see above

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:166]: add an enum field with the options

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:171]:
```suggestion
                    "description": "The BUSCO lineage to use, or `auto` to automatically select lineage"
```

File: nextflow_schema.json
Comment (mashehu) [inline nextflow_schema.json:230]:
```suggestion
                    "description": "Override library type inferred based on strandedness defined in meta object. `A` for auto."
```

File: subworkflows/local/utils_nfcore_denovotranscript_pipeline/main.nf
Comment (mashehu) [inline subworkflows/local/utils_nfcore_denovotranscript_pipeline/main.nf:160]:
```suggestion
        error("Incompatible parameters: cannot use --skip_assembly and --qc_only modes together.")
```

**Review by mashehu — APPROVED** [review]
some smaller things left. rest looks good! nice job!

File: conf/modules.config
Comment (fbdtemme) [inline conf/modules.config:22]: I think this can be simplified to just a single `.*`

File: conf/modules.config
Comment (fbdtemme) [inline conf/modules.config:32]: Same here

**Review by fbdtemme — APPROVED** [review]
Looks good to me as well! Just one more small suggestion.

File: CITATIONS.md
Comment (avani-bhojwani) [reply CITATIONS.md:1]: Yes, I've made this change

File: docs/images/mqc_general_stats.png
Comment (avani-bhojwani) [reply docs/images/mqc_general_stats.png:1]: Thank you, I have updated my multiqc report. I have reduced it to 3 lines per sample: 
- Since we're working with paired-end reads, some stats will need to be split into the left and right read (2 lines)
- Some stats are for the sample overall (1 line)

File: docs/usage.md
Comment (avani-bhojwani) [reply docs/usage.md:95]: Thanks, I've split it into two sentences.

File: docs/usage.md
Comment (avani-bhojwani) [reply docs/usage.md:128]: This param is coming from rnaSPAdes, so I will leave it as `ss`

File: modules/local/orp_transrate/main.nf
Comment (avani-bhojwani) [reply modules/local/orp_transrate/main.nf:1]: yes, I would like to do this in the future

File: modules/local/rnaquast/main.nf
Comment (avani-bhojwani) [reply modules/local/rnaquast/main.nf:1]: Hoping to add to nf-core modules, but will need more time to do it

File: nextflow.config
Comment (avani-bhojwani) [reply nextflow.config:38]: same as before (param is from rnaSPAdes)

**Pull request merged**
